Frameworks/Foundation/NSURLProtocol.mm

```diff
@@ -46,20 +46,20 @@ + (BOOL)registerClass:(id)cls {
     }
 }
 
-+ (id)_URLProtocolClassForRequest:(id)request {
++ (Class)_URLProtocolClassForRequest:(NSURLRequest*)request {
     @synchronized(self) {
-        NSArray* classes = [self _registeredClasses];
+        NSArray<Class>* classes = [self _registeredClasses];
         NSInteger count = [classes count];
 
         while (--count >= 0) {
-            id check = [classes objectAtIndex:count];
+            Class check = [classes objectAtIndex:count];
```

Comment:
@aballway This is iterating in reverse. Cheaper than a reverseObjectEnumerator,.